sync/sync.go

```diff
@@ -0,0 +1,85 @@
+package sync
+
+import (
+	"sync"
+)
+
+// Bool type with concurrent access support.
+type Bool struct {
+	mu    sync.RWMutex
```

Comment:
You can also use the namining convension `rw` for RWMutex if it sounds better to you.
https://golang.org/pkg/sync/#RWMutex